implementation/src/main/java/io/smallrye/config/Converters.java

```diff
@@ -562,26 +566,44 @@ public A convert(final String str) {
 
     static final class BuiltInConverter<T> implements Converter<T>, Serializable {
         private final int id;
+        private final StringCleanup stringCleanup;
         private final Converter<T> function;
 
         static <T> BuiltInConverter<T> of(int id, Converter<T> function) {
-            return new BuiltInConverter<>(id, function);
+            return new BuiltInConverter<>(id, StringCleanup.NONE, function);
+        }
+
+        static <T> BuiltInConverter<T> of(int id, StringCleanup stringCleanup, Converter<T> function) {
+            return new BuiltInConverter<>(id, stringCleanup, function);
         }
 
-        private BuiltInConverter(final int id, final Converter<T> function) {
+        private BuiltInConverter(final int id, final StringCleanup stringCleanup, final Converter<T> function) {
             this.id = id;
+            this.stringCleanup = stringCleanup;
             this.function = function;
         }
 
         public T convert(final String value) {
-            return function.convert(value);
+            final String cleanValue;
```

Comment:
Maybe use a ternary operator instead? Then a new variable doesn't need to be allocated